### PR TITLE
feat: multiple base features to allow offline use cases

### PR DESCRIPTION
**Pull request by JanZachmann**
- web service:
  - refactoring in order to allow coexistence with twin
  - added routes to trigger republish and status
- reboot: added configuration of wait-online reboot timeout
- os-release path: change to /usr/lib/os-release
- factory-reset: improved restore settings path validation

The coresponding omnect-ui PR: https://github.com/omnect/omnect-ui/pull/1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -320,6 +324,46 @@ In all other cases there will be an error status:
 }
 ```
 
+#### Configure wait-online reboot timeout
```
Comment (mlilien) [inline README.md:327]: do we want to document, that we test for `--any` per default and `OMNECT_WAIT_ONLINE_INTERFACES` can alter this configuration? resp. provide this as optional parameter in this method?

File: testfiles/negative/systemd-networkd-wait-online-invalid-default-timeout.service [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+DefaultDependencies=no
+Conflicts=shutdown.target
+Requires=systemd-networkd.service
+After=systemd-networkd.service
+Before=network-online.target shutdown.target
+
+[Service]
+Type=oneshot
+EnvironmentFile=-testfiles/negative/systemd-networkd-wait-online-invalid-timeout.env
+ExecStart=/bin/bash -c '/lib/systemd/systemd-networkd-wait-online ${OMNECT_WAIT_ONLINE_INTERFACES_RUN:---any --interface=eth0 --interface=wlan0 } --timeout=${OMNECT_WAIT_ONLINE_TIMEOUT_IN_SECS:-300}'
```
Comment (mlilien) [inline testfiles/negative/systemd-networkd-wait-online-invalid-default-timeout.service:14]: `OMNECT_WAIT_ONLINE_INTERFACES_RUN` got renamed to
`OMNECT_WAIT_ONLINE_INTERFACES` and the default config is meanwhile only `--any`.
i guess this is not really relevant for the test, but if at some point somebody wonders.

File: README.md (hunk 1 above)
Comment (JanZachmann) [reply README.md:327]: > do we want to document, that we test for `--any` per default and OMNECT_WAIT_ONLINE_INTERFACES can alter this configuration?

yes

> resp. provide this as optional parameter in this method?

would this really be a use case? I propose to create a "to be discussed task"

**Review by mlilien — APPROVED** [review]

File: src/systemd/wait_online.rs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            .unwrap_or(SERVICE_FILE_PATH_DEFAULT.to_string())
+    }};
+}
+
+macro_rules! env_file_path {
+    () => {{
+        const ENV_FILE_PATH_DEFAULT: &'static str = "/etc/omnect/systemd-networkd-wait-online.env";
+        std::env::var("ENV_FILE_PATH").unwrap_or(ENV_FILE_PATH_DEFAULT.to_string())
+    }};
+}
```
Comment (empwilli) [inline src/systemd/wait_online.rs:20]: Why as separate macros and not via lazy_static?

**Review by empwilli — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -687,27 +731,76 @@ The availability of the feature is reported by the following module twin propert
...
 
-omnect-device-service provides a http server that exposes a web API over a unix domain socket.<br>
-Information about the socket can be found in the appropriate [socket file](systemd/omnect-device-service.socket)
+The web service features is disabled by default and must be explicitly activated via environment variable `WEBSERVICE_ENABLED=true`.
 
 ### Trigger reboot
 
 ```
-curl -X PUT --unix-socket /run/omnect-device-service/api.sock http://localhost/reboot
+curl -X POST --unix-socket /run/omnect-device-service/api.sock http://localhost/reboot/v1
```
[the remark this answers is not in the document]
Comment (JanZachmann) [reply README.md:744]: each route is versioned separately

File: src/systemd/wait_online.rs (hunk 3 above)
Comment (JanZachmann) [reply src/systemd/wait_online.rs:20]: the reason is testability: in some testcases we change the file paths. this doesn't work with lazy_static. I couldn't find another solution yet.

**Pull request merged**